tracing-opentelemetry/src/tracer.rs

```diff
@@ -0,0 +1,96 @@
+use opentelemetry::{api, sdk};
+
+/// A protocol for OpenTelemetry [`Tracer`]s that are capable of producing
+/// sampled span contexts _before_ starting their associated spans.
+///
+/// This enables interoperability between `tracing` and `opentelemetry` by
+/// allowing otel trace ids to be associated _after_ a `tracing` span has been
+/// created. See the [`OpenTelemetrySpanExt`] in this crate for usage examples.
+///
+/// [`Tracer`]: https://docs.rs/opentelemetry/latest/opentelemetry/api/trace/tracer/trait.Tracer.html
+/// [`OpenTelemetrySpanExt`]: trait.OpenTelemetrySpanExt.html
+pub trait PreSampledTracer {
```

Comment:
Do we expect users to implement this trait for other types implementing OpenTelemetry's `Tracer` trait? If so, does it make sense to have an example of how it might be implemented, or something?

Otherwise, if we _don't_ expect users to implement this, it should probably be sealed or something.

Reply:
This would be implemented by someone who wanted an alternate implementation of the opentelemetry sdk, which is possible, but none exist. This isn't really meant to be used by users of tracing directly.